include/eve/traits/overload/default_behaviors.hpp

```diff
@@ -0,0 +1,157 @@
+//======================================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//======================================================================================================================
+#pragma once
+
+#include <eve/concept/value.hpp>
+#include <eve/detail/skeleton.hpp>
+#include <eve/detail/function/conditional.hpp>
+
+namespace eve
+{
+  namespace detail
+  {
+    // Usual object function so we can have EVE_FORCEINLINE there for QoI on ARM/AARCH64
+    inline struct { EVE_FORCEINLINE auto operator()(auto, auto x) const { return x; } } eval_2nd = {};
+  }
+
+  //====================================================================================================================
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct callable : decorated_with<OptionsValues, Options...>
+  {
+    using base = decorated_with<OptionsValues, Options...>;
+
+    template<typename T>
+    auto operator[](T t) const
+    requires( requires(base const& b) { b[t];} )
+    {
+      auto new_traits = base::operator[](t);
+      return  Func<decltype(new_traits)>{new_traits};
+    }
+
+    template<typename T> void operator[](T t) const =delete;
+
+    template<typename... Args>
+    constexpr auto behavior(auto arch, Args&&... args) const
+    {
+      return Func<OptionsValues>::deferred_call(arch, EVE_FWD(args)...);
+    }
+  };
+
+  //====================================================================================================================
+
+  //====================================================================================================================
+  template< template<typename> class Func
+          , typename OptionsValues
+          , typename... Options
+          >
+  struct elementwise_callable : callable<Func, OptionsValues, conditional_option, Options...>
+  {
+    Func<OptionsValues> const& current() const { return static_cast<Func<OptionsValues>>(*this); }
```

Comment:
I thought it's usually called `self` or `derived_self`.
Aslo - maybe put this into `callable`?
Also - constexpr

Reply:
derived, and pushed back in callable as a protected member